src/Homie/Boot/BootConfig.cpp

```diff
@@ -339,25 +341,25 @@ void BootConfig::_onDeviceInfoRequest(AsyncWebServerRequest *request) {
   Interface::get().getLogger() << F("Received device information request") << endl;
   auto numSettings = IHomieSetting::settings.size();
   auto numNodes = HomieNode::nodes.size();
-  DynamicJsonBuffer jsonBuffer(JSON_OBJECT_SIZE(5) + JSON_OBJECT_SIZE(2) + JSON_ARRAY_SIZE(numNodes) + (numNodes * JSON_OBJECT_SIZE(2)) + JSON_ARRAY_SIZE(numSettings) + (numSettings * JSON_OBJECT_SIZE(5)));
-  JsonObject& json = jsonBuffer.createObject();
+  DynamicJsonDocument jsonDocument(JSON_OBJECT_SIZE(5) + JSON_OBJECT_SIZE(2) + JSON_ARRAY_SIZE(numNodes) + (numNodes * JSON_OBJECT_SIZE(2)) + JSON_ARRAY_SIZE(numSettings) + (numSettings * JSON_OBJECT_SIZE(5)));
+  JsonObject json = jsonDocument.as<JsonObject>();
   json["hardware_device_id"] = DeviceId::get();
   json["homie_esp8266_version"] = HOMIE_ESP8266_VERSION;
-  JsonObject& firmware = json.createNestedObject("firmware");
+  JsonObject firmware = json.createNestedObject("firmware");
   firmware["name"] = Interface::get().firmware.name;
   firmware["version"] = Interface::get().firmware.version;
 
-  JsonArray& nodes = json.createNestedArray("nodes");
+  JsonArray nodes = json.createNestedArray("nodes");
   for (HomieNode* iNode : HomieNode::nodes) {
-    JsonObject& jsonNode = jsonBuffer.createObject();
+    JsonObject jsonNode = jsonDocument.as<JsonObject>();
     jsonNode["id"] = iNode->getId();
     jsonNode["type"] = iNode->getType();
     nodes.add(jsonNode);
   }
 
-  JsonArray& settings = json.createNestedArray("settings");
+  JsonArray settings = json.createNestedArray("settings");
   for (IHomieSetting* iSetting : IHomieSetting::settings) {
-    JsonObject& jsonSetting = jsonBuffer.createObject();
+    JsonObject jsonSetting = jsonDocument.as<JsonObject>();
```

Comment:
With `jsonDocument` not initialized this will lead to undefined behaviour as described above. With `jsonDocument` initialized, this will put the attributes of the last setting into the root of `jsonDocument`.
Use `JsonObject jsonSetting = settings.createNestedObject();` instead and remove the add command below.